### ISSUE TITLE
Provide SASL/SCRAM based client-authentication for aws_msk_cluster

### ISSUE THREAD

**Issue by dishantkamble** [hashicorp/terraform-provider-aws#15298]
### Community Note 

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In addition to `client_authentication` possible via `tls` there is no option provided for SASL/SCRAM based `client_authentication` in `aws_msk_cluster` resource. Additionally, there are no/selective resources available for the list of operations possible for kafka (AWS MSK) as mentioned in the reference below.

I am specifically looking for `batch-associate-scram-secret` operation to enable SASL/SCRAM based authentication via terraform.

### New or Affected Resource(s)

* aws_msk_cluster
* aws_msk_configuration

### Potential Terraform Configuration

Possible option for SASL/SCRAM based authentication can be as follows for the client_authentication Argument Reference

 ```
resource "aws_msk_cluster" "example" {
  cluster_name           = "example"
  kafka_version          = "2.4.1"
  number_of_broker_nodes = 3

  broker_node_group_info {
    instance_type   = "kafka.m5.large"
    ebs_volume_size = 1000
    client_subnets = [
      aws_subnet.subnet_az1.id,
      aws_subnet.subnet_az2.id,
      aws_subnet.subnet_az3.id,
    ]
    security_groups = [aws_security_group.sg.id]
  }

  client_authentication {
    sasl {
      secretArnList = [aws_secretsmanager_secret.example-1.arn, aws_secretsmanager_secret.example-2.arn]
    }
  }

  encryption_info {
    encryption_at_rest_kms_key_arn = aws_kms_key.kms.arn
  }

  open_monitoring {
    prometheus {
      jmx_exporter {
        enabled_in_broker = true
      }
      node_exporter {
        enabled_in_broker = true
      }
    }
  }
```

### References

* [AWS Kafka Commands](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/kafka/index.html#cli-aws-kafka)

**Comment by ewbankkit** [comment]
Related (enabling SASL/SCRAM on `aws_msk_cluster`):

* https://github.com/terraform-providers/terraform-provider-aws/issues/15237

The SCRAM secret will need to be a new resource.
It looks like #15302 will address both.

**Comment by PavelDemyanenko** [comment]
Any news so far?

**Comment by anGie44** [comment]
Hi @dishantkamble et al. 👋 dropping a note here that we've just merged in (PR #15302) sasl/scram authentication support in the `msk_cluster` resource in addition to adding a new resource, namely `aws_msk_scram_secret_association` for associating secrets with an MSK cluster. These additions will be in `v3.18.0` of the Terraform AWS Provider, expected out later today.
